### DISCUSSION TITLE
section 3.14 - Erro para criação do Bucket

### DISCUSSION THREAD

**Discussion by chwiee** [chgasparoto/curso-aws-com-terraform#113] [Q&A]
Na hora que eu rodo o terraform para criar o bucket e subir um arquivo ele da o seguinte erro:

```

Changes to Outputs:
  + bucket-arn    = (known after apply)
  + bucket-domain = (known after apply)
aws_s3_bucket.this: Creating...
╷
│ Error: error creating S3 Bucket (MyHero-backend-dev): InvalidBucketName: The specified bucket is not valid.
│       status code: 400, request id: 5Z3D8D28RK2CDH4F, host id: K5o4kkgXTDeb51/QdAuF/At2046NdlCke2MOdXBCT7gqW7WGrv2spH4Sufni6N4L8q48aYVhje0= 
│
│   with aws_s3_bucket.this,
│   on main.tf line 1, in resource "aws_s3_bucket" "this":
│    1: resource "aws_s3_bucket" "this" {
│
╵
```

O nome do bucket não existe pois criei ele na mão (e logo após apaguei) o que eu percebi é que ele roda na ordem inversa, ele tenta subir o arquivo antes mesmo de criar o Bucket, se eu rodo o comando em partes por exemplo, primeiro coloco apenas o resource de criação e logo após rodo com os demais resources ele cria e sobe o arquivo normalmente, pode me ajduar?

**Comment by chgasparoto** [comment]
@chwiee o erro que você postou está falando que o nome do bucket está inválido e em fato realmente está. Você não pode ter letra maiúscula. Mais detalhes [aqui](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html).

Em relação a ordem de criação do recursos, se você referenciar o bucket dentro do recurso do objeto o Terraform sabe que deve criar primeiro o bucket para depois criar o objeto.

**Reply by chwiee** [reply]
Opa mestre vlw, eu realmente não sabia que não pode ter letras Maiusculas no nome do bucket =D

vlw!
